### DISCUSSION TITLE
Route audio output of SDR++ to other apps?

### DISCUSSION THREAD

**Discussion by pmaine** [AlexandreRouma/SDRPlusPlus#256] [Q&A]
I am running SDR++ on a Raspberry Pi 4B on the Raspian OS. How can I route the audio out of  SDR++ into another app such as FLDIGI or QSSTV?

**Comment by starvald** [comment]
you can add a virtual sink to the /etc/pulse/default.pa config.
add to the end of the config.

`load-module module-null-sink sink_name=VirtualSink1 sink_properties=device.description=VirtualSink1`

save then reboot.

then install pavucontrol you can then set sdrpp audio and qsstv to the virtual sink.

you can add as many virtual sinks as you need.

**Reply by pmaine** [reply]
Hi Lee

I appreciate all of the assistance!

I followed your steps exactly and when I rebooted the menu bar at the top of the screen disappeared after I rebooted. I commented our your lines and rebooted and the menu reappeared. Any ideas?

**Reply by starvald** [reply]
very strange, pulseaudio shouldn't anything to do with the gui.

this is what i have at the end of my default.pa without issue.

```
# Added line for virtual sink
load-module module-null-sink sink_name=VirtualSink1 sink_properties=device.description=VirtualSink1
load-module module-null-sink sink_name=VirtualSink2 sink_properties=device.description=VirtualSink2
load-module module-null-sink sink_name=VirtualSink3 sink_properties=device.description=VirtualSink3
load-module module-null-sink sink_name=VirtualSink4 sink_properties=device.description=VirtualSink4
```

**Reply by pmaine** [reply]
Lee
I incorrectly added separate lines per your original post. I put the commands on one line. I next rebooted and the virtual device was added AND my menu is normal.

I can now route audio from SDR++ to QSSTV!

thanks again!

**Comment by the-gobleo-tinkerer** [comment]
For just routing the audio into other apps, I think `pavucontrol` will suffice. Open the program that is to receive audio (so that it shows up in pavucontrol), go to the Recording tab, and change the source to "Monitor of [whatever the Pi's audio output is named]". Make sure you have enabled audio output in `raspi-config` so that you do have an audio output device.

![Screenshot from 2021-08-15 08-52-15](https://user-images.githubusercontent.com/41851763/129463645-841b09b1-2783-42c9-9824-dda269144507.png)

**Reply by pmaine** [reply]
It’s working now!
Thank you

**Reply by starvald** [reply]
> For just routing the audio into other apps, I think `pavucontrol` will suffice. Open the program that is to receive audio (so that it shows up in pavucontrol), go to the Recording tab, and change the source to "Monitor of [whatever the Pi's audio output is named]". Make sure you have enabled audio output in `raspi-config` so that you do have an audio output device.
> 

true, you don't need to add virtualsinks but it does mean you can route audio to apps without hearing it while still being able to hear a vfo you want to listen to.

**Reply by pmaine** [reply]
Hi Lee
I’m interested on how you use your virtual audio devices.  I went ahead and added four virtual devices like you. Please share!

**Reply by starvald** [reply]
for the 20m band i usually have fldigi for psk, 2 for SSTV and either one for Olivia mode or ft8.
11m band js8call, ROS and SSTV.

**Reply by pmaine** [reply]
Git won't allow me to mark as answered.

**Comment by guitarpicva** [comment]
I found [https://craigwilson.blog/post/2024/2024-12-05-virtualaudiosplit/](url) helpful.  Only the creation of the virtual sinks.  Then it was a matter of selecting the sink in the SDR++ sink config.

FLDIGI or other sound card program would then listen on the MONITOR of that virtual sink, as found in pulse audio or pipewire tools.

This method is a user level configuration as well, living in the ~/.config/ directory structure.
